### PR TITLE
Rename usernames to nicknames and limit to 15 chars

### LINKED ISSUE #7: Limit usernames to 15 characters
Also rename "pick a username" to "pick a nickname" to encourage brevity.

### PR DESCRIPTION
**Pull request by vinceau**
Fixes #7

### CODE REVIEW

File: web500/templates/login.html [hunk 1]
```diff
@@ -3,14 +3,14 @@
 {% block content %}
   <h1>welcome to web500</h1>
   {% if invalid_login %}
-  <p>Username is taken</p>
+  <p>Nickname is taken or is too long (max: 15 chars).</p>
```
Comment (nathyong) [inline web500/templates/login.html:6]: A little bit lazy :P but we can fix this up later with the change to per-room usernames.

File: web500/routes.py [hunk 2]
```diff
@@ -20,11 +32,10 @@ def login():
         return redirect(url_for("index"))
 
     if request.method == "POST":
-        if not query_db("select * from users where username = ?",
-                        [request.form['username']]):
-            session['username'] = request.form['username']
-            update_db("insert into users (username) values (?)",
-                      [request.form['username']])
+        username = request.form['username']
+        if _is_valid(username) and not query_db("select * from users where username = ?", [username]):
```
Comment (nathyong) [inline web500/routes.py:36]:
```
if _is_valid(username) \
    and not query_db("select....."):
```

Comment (vinceau) [inline web500/routes.py:36]: I contemplated that... but I don't think that it's... nice. Your call.

Comment (nathyong) [inline web500/routes.py:36]: We will most certainly rewrite this line when we change the username structure though.

**Pull request merged**